engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/AuthenticationProvider.scala

```diff
@@ -19,12 +20,11 @@ trait AuthenticationProvider {
 
 object AuthenticationProvider extends LazyLogging {
   def apply(config: Config, classLoader: ClassLoader, allCategories: List[String])(implicit ec: ExecutionContext, sttpBackend: SttpBackend[Future, Nothing, NothingT]): AuthenticationProvider = {
-    val loaded = ScalaServiceLoader.loadClass[AuthenticationProvider](classLoader) {
-      AuthenticationConfiguration.parseMethod(config) match {
-        case AuthenticationMethod.OAuth2 => OAuth2AuthenticationProvider()
-        case _ => BasicAuthenticationProvider()
-      }
-    }
+    val loaded = ScalaServiceLoader.loadNamed[AuthenticationProvider](
+      config.getString(AuthenticationConfiguration.methodConfigPath),
+      classLoader,
+      List(OAuth2AuthenticationProvider(), BasicAuthenticationProvider())
```

Comment:
I wonder - do we need these defaults? Can't we just register those providers with ServiceLoader?

Reply:
I have replaced the defaults with a services file.